lib/propolis/src/hw/nvme/mod.rs

```diff
@@ -519,13 +523,21 @@ impl PciNvme {
         let mut sn: [u8; 20] = [0u8; 20];
         sn[..sz].clone_from_slice(&serial_number.as_bytes()[..sz]);
 
+        // If MDTS is specified, ensure that it is either 0 (meaning no limit)
+        // or is at least as large as a single page.
+        let mdts = match mdts {
+            None | Some(0) => 0,
+            Some(v) => v.max(PAGE_SHIFT as u8),
+        };
```

Comment:
Since we're hardcoding the arg in all call sites right now can we just move it into the body for now? As-is I think it can be a bit hard to get correct and ensure a valid value is given. Case in point, 20 means we're saying a max size of `2^20 * CAP.MPSMIN (4K) = 4GiB` and not the intended `1MiB` 😅️

(Also, should that `max` have been a `min`?)

Reply:
Yeah, this is wrong.  OVMF in particular freaks out, since it uses a u32 for calculating the max transfer size.  I'll be posting an updated wad shortly.

Reply:
The obvious math bug (deviating from the spec) should be fixed now.